### DISCUSSION TITLE
Questions related to border options

### DISCUSSION THREAD

**Discussion by SmiteshP** [MunifTanjim/nui.nvim#248] [Q&A] [answered]
Hi @MunifTanjim !
Thanks for writing this amazing library, it makes working with UI so much simpler. While working on my [this](https://github.com/SmiteshP/nvim-navbuddy) plugin nvim-navbuddy. I noticed a few bugs.

1) On the popup window with `text`, it seems to be using "Normal" highlights for highlighting the borders. Thats why I had to map Normal to FloatBorder as a workaround [here](https://github.com/SmiteshP/nvim-navbuddy/blob/master/lua/nvim-navbuddy/display.lua#L261). I suspect [this issue](https://github.com/SmiteshP/nvim-navbuddy/issues/3) opened on my plugins is caused by this.

2) In my plugin I have written some logic for fusing of borders of neighboring popup windows. For these I am passing the border_chars manually. But [here](https://github.com/SmiteshP/nvim-navbuddy/blob/master/lua/nvim-navbuddy/display.lua#L195-L204) I had to directly pass the list of border_chars (`{ <border_chars_here>}`) whereas the documentation suggests to have table like  `{ style = { <border_chars_here> } }`. This also seems to be somehow linked to this window having text on its border. Passing the table as per documentation causes errors.

3) [Feature request] It would be very nice if this "fusing" of borders for adjacent popups be a option within nui itself.

**Comment by MunifTanjim** [comment, marked as answer]
> On the popup window with text, it seems to be using "Normal" highlights for highlighting the borders. Thats why I had to map Normal to FloatBorder as a workaround [here](https://github.com/SmiteshP/nvim-navbuddy/blob/master/lua/nvim-navbuddy/display.lua#L261). I suspect https://github.com/SmiteshP/nvim-navbuddy/issues/3 opened on my plugins is caused by this.

You might just want to use `"Normal:Normal,FloatBorder:NavbuddyFloatBorder"` as `winhighlight` for all the popups.

- `FloatBorder` is used for popup border.
- `Normal` is used for the popup body.

If you use `"Normal:NavbuddyFloatBorder,FloatBorder:NavbuddyFloatBorder"`, then highlight for popup's body and border become the same.

---

If you want the border text to have the same color as border, you can replace https://github.com/SmiteshP/nvim-navbuddy/blob/b043eef7a42aa9d13618e59eb6a13ebd82ee631a/lua/nvim-navbuddy/display.lua#L244 with:

```lua
		bottom = NuiText("["..obj.lsp_name.."]", "FloatBorder"),
```

**Comment by MunifTanjim** [comment]
> In my plugin I have written some logic for fusing of borders of neighboring popup windows. For these I am passing the border_chars manually. But [here](https://github.com/SmiteshP/nvim-navbuddy/blob/master/lua/nvim-navbuddy/display.lua#L195-L204) I had to directly pass the list of border_chars ({ <border_chars_here>}) whereas the documentation suggests to have table like { style = { <border_chars_here> } }. This also seems to be somehow linked to this window having text on its border. Passing the table as per documentation causes errors.

- https://github.com/SmiteshP/nvim-navbuddy/blob/b043eef7a42aa9d13618e59eb6a13ebd82ee631a/lua/nvim-navbuddy/display.lua#L195-L204

You need to do this because you're passing the return value to `border.style` for the right popup

- https://github.com/SmiteshP/nvim-navbuddy/blob/b043eef7a42aa9d13618e59eb6a13ebd82ee631a/lua/nvim-navbuddy/display.lua#L256-L257

But for the other popups you're passing the return value to `border`:

- https://github.com/SmiteshP/nvim-navbuddy/blob/b043eef7a42aa9d13618e59eb6a13ebd82ee631a/lua/nvim-navbuddy/display.lua#L234
- https://github.com/SmiteshP/nvim-navbuddy/blob/b043eef7a42aa9d13618e59eb6a13ebd82ee631a/lua/nvim-navbuddy/display.lua#L223

Passing the table as per documentation won't cause any error. You just need to adjust your code to make it consistent. Always pass the return value to `border` or always pass to `border.style` and then you can keep the return value of `get_border_chars` function consistent.

**Comment by SmiteshP** [comment]
```lua
bottom = NuiText("["..obj.lsp_name.."]", "FloatBorder"),
```

This worked, Thanks 👍🏽 

> You need to do this because you're passing the return value to border.style for the right popup

Oh right! My bad 🤦‍♂️
Should have figured this out myself.
